src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComActivator.cs

```diff
@@ -326,7 +388,97 @@ public void RequestLicKey(int dwReserved, [MarshalAs(UnmanagedType.BStr)] out st
                 [MarshalAs(UnmanagedType.BStr)] string bstrKey,
                 [MarshalAs(UnmanagedType.Interface)] out object ppvObject)
             {
-                throw new NotImplementedException();
+                Debug.Assert(pUnkReserved == null);
+                this.CreateInstanceInner(pUnkOuter, ref riid, bstrKey, isDesignTime: false, out ppvObject);
+            }
+
+            private void CreateInstanceInner(
+                object pUnkOuter,
+                ref Guid riid,
+                string key,
+                bool isDesignTime,
+                out object ppvObject)
+            {
+                BasicClassFactory.ValidateInterfaceRequest(this.classType, ref riid, pUnkOuter);
+
+                ppvObject = this.licenseProxy.AllocateAndValidateLicense(this.classType, key, isDesignTime);
+                if (pUnkOuter != null)
+                {
+                    ppvObject = BasicClassFactory.CreateAggregatedObject(pUnkOuter, ppvObject);
+                }
+            }
+        }
+
+        private class LicenseInteropProxy
+        {
+            private static readonly Type s_LicenseAttrType;
+            private readonly Type helperType;
+            private readonly MethodInfo getLicInfo;
+            private readonly MethodInfo requestLicKey;
+            private readonly MethodInfo allocateAndValidateLicense;
+            private readonly object instance;
+
+            static LicenseInteropProxy()
+            {
+                s_LicenseAttrType = Type.GetType("System.ComponentModel.LicenseProviderAttribute, System", throwOnError: true);
+            }
+
+            public LicenseInteropProxy()
+            {
+                Type licman = Type.GetType("System.ComponentModel.LicenseManager, System", throwOnError: true);
+                this.helperType = licman.GetNestedType("LicenseInteropHelper", BindingFlags.NonPublic);
```

Comment:
Nit: We use this only when absolutely required to make the code work

Reply:
Agreed. This is related to the mop-up I was planning on doing post-checkin. I can fix this now, but it will make the PR much larger since I would want to remove broken code. I want to get in support and then remove the nasty bits - like this - from both corefx and coreclr.